core/domain/exp_services.py

```diff
@@ -950,21 +953,32 @@ def update_exploration(
     exploration = apply_change_list(exploration_id, change_list)
     _save_exploration(committer_id, exploration, commit_message, change_list)
 
-    # Update summary of changed exploration.
-    update_exploration_summary(exploration.id)
 
-
-def create_exploration_summary(exploration_id):
+def create_exploration_summary(exploration_id, committer_id=None):
     """Create summary of an exploration and store in datastore."""
     exploration = get_exploration_by_id(exploration_id)
     exp_summary = compute_summary_of_exploration(exploration)
+    # update the contributor id list if necessary (contributors
+    # defined as humans who have made a positive (i.e. not just
+    # a revert) change to an exploration's content)
+    if committer_id is not None:
+        if committer_id not in feconf.SYSTEM_USER_IDS:
+            exp_summary.contributor_ids = [committer_id]
     save_exploration_summary(exp_summary)
 
 
-def update_exploration_summary(exploration_id):
+def update_exploration_summary(exploration_id, committer_id=None):
     """Update the summary of an exploration."""
     exploration = get_exploration_by_id(exploration_id)
     exp_summary = compute_summary_of_exploration(exploration)
+    # update the contributor id list if necessary (contributors
+    # defined as humans who have made a positive (i.e. not just
+    # a revert) change to an exploration's content)
+    if committer_id is not None:
+        if committer_id not in feconf.SYSTEM_USER_IDS:
+            contributor_id_set = set(exp_summary.contributor_ids)
```

Comment:
Feels like lots of unnecessary conversions. How about the following logic instead: if the committer_id isn't in the list, append it?